exploration/syntax-exploration-2.md

```diff
@@ -0,0 +1,256 @@
+  for quoted patterns.
+
+> The use in the previous iteration of `{{` and `}}` surrounding code-mode messages was identified as a potential negative, 
+> as the closing `}}` are superfluous to the message.
+
+The use of `>>` to represent the "starting code-mode sigil" is **_not_** final.
+**_Do not fixate on the specific character sequence when choosing (or not) this design._**
+
+**Examples**
+```
+Hello world!
+```
+```
+Hello {$user}
+```
+```
+>>input {$user :function option=value}
+{Hello {$user}}
+
+>>input {$user :function option=value}{Hello {$user}}
+```
+```
+>>input {$var :function option=value}
+local $foo = {$bar :function option=value}
+{Hello {$var}, you have a {$foo}}
+
+>>input {$var :function option=value}local $foo = {$bar :function option=value}{Hello {$var}, you have a {$foo}}
+```
+```
+>>match {$foo}
+when foo {Hello {$foo} you have a {$var}}
+when * {{$foo} hello you have a {$var}}
+
+>>match {$foo}when foo{Hello {$foo} you have a {$var}}when *{{$foo} hello you have a {$var}}
+```
+
+```
+>>match {$foo :function option=value} {$bar :function option=value}
+when a b {  {$foo} is {$bar}  }
+when x y {  {$foo} is {$bar}  }
+when * * {  {$foo} is {$bar}  }
+
+>>match {$foo :function option=value}{$bar :function option=value}when a b{  {$foo} is {$bar}  }when x y{  {$foo} is {$bar}  }when * *{  {$foo} is {$bar}  }
+```
+
+### Candidate 3a
+
+**Description:** Use sigils for code mode, use `{`/`}` for keys
+
+- Starts in text mode. 
+- Uses a sigil for code statements. 
+  - Requires the code-mode sigil to be escaped in **_unquoted_** patterns.
+  - Quoted patterns do not require this sigil to be escaped.
+  - Note that decisions about auto-trimming affect the escaping situation. 
+    If quoting the pattern is required, no additional escapes are needed.
+- Placeholders and expressions use `{`/`}`:
+  > `{$var}`, `{unquoted}`, `{|quoted|}`, `{$var :function}`
+
+The use of `%` as the sigil is **_not_** an inherent part of the design here.
+The actual sigil or sigil sequence needs to be decided. 
+Note that using a doubled-sigil reduces the need for escaping.
+
+Patterns in this syntax are generally unquoted,
+but MAY be quoted with the 2023-10-09 consensus `{{`/`}}` quotes.
+
+Patterns lacking declarations, selectors, or both, are **_not_** auto-trimmed.
+Whitespace in quoted patterns is significant.
+Whitespace in unquoted patterns may or may not be significant (hot topic).
+
+```
+Hello world!
+```
+```
+Hello {$user}!
+```
+```
+%input {$var :function option=value}
+Hello {$var}
+
+%input {$var :function option=value}Hello {$var}
+```
+```
+%input {$var :function option=value}
+%local $foo = {$bar :function option=value}
+Hello {$var}, you have a {$foo}
+
+%input {$var :function option=value}%local $foo = {$bar :function option=value}Hello {$var}, you have a {$foo}
+```
+```
+%match {$foo}
+%when{foo} Hello {$foo} you have a {$var}
+%when{*} {$foo} hello you have a {$var}
```

Comment:
Or even:

```suggestion
%[when foo] Hello {$foo} you have a {$var}
%[when *] {$foo} hello you have a {$var}
```

This would make `%` safe to use again, and only require escaping `%[`.